cmd/remote_enforcer.go

```diff
@@ -0,0 +1,291 @@
+package main
+
+import (
+	"container/list"
+	"crypto/ecdsa"
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"os/user"
+	"strconv"
+	"sync"
+	"time"
+
+	log "github.com/Sirupsen/logrus"
+	"github.com/aporeto-inc/trireme/cache"
+	"github.com/aporeto-inc/trireme/collector"
+	"github.com/aporeto-inc/trireme/enforcer"
+	_ "github.com/aporeto-inc/trireme/enforcer/utils/nsenter"
+	"github.com/aporeto-inc/trireme/enforcer/utils/packet"
+	"github.com/aporeto-inc/trireme/enforcer/utils/rpcwrapper"
+	"github.com/aporeto-inc/trireme/enforcer/utils/tokens"
+	"github.com/aporeto-inc/trireme/policy"
+	"github.com/aporeto-inc/trireme/supervisor"
+	"github.com/aporeto-inc/trireme/supervisor/iptablesutils"
+	"github.com/aporeto-inc/trireme/supervisor/provider"
+)
+
+const (
+	ipcProtocol         = "unix"
+	defaultPath         = "/var/run/default.sock"
+	statsContextID      = "UNUSED"
+	defaulttimeInterval = 2
+	envStatsChannelPath = "STATSCHANNEL_PATH"
+	envSocketPath       = "SOCKET_PATH"
+)
+
+//CollectorImpl exported
+type CollectorImpl struct {
+	Flowentries *list.List
+	sync.RWMutex
+}
+type collectorentry struct {
+	L4FlowHash string
+	entry      *enforcer.StatsPayload
+}
+
+//CollectFlowEvent expoted
+func (c *CollectorImpl) CollectFlowEvent(contextID string, tags policy.TagsMap, action string, mode string, sourceID string, tcpPacket *packet.Packet) {
+	l4FlowHash := tcpPacket.L4FlowHash()
+	payload := &enforcer.StatsPayload{ContextID: contextID,
+		Tags:   tags,
+		Action: action,
+		Mode:   mode,
+		Source: sourceID,
+		Packet: tcpPacket,
+	}
+
+	c.Lock()
+	c.Flowentries.PushBack(&collectorentry{L4FlowHash: l4FlowHash, entry: payload})
+	c.Unlock()
+}
+
+//CollectContainerEvent exported
+//This event should not be expected here in the enforcer process inside a particular container context
+func (c *CollectorImpl) CollectContainerEvent(contextID string, ip string, tags policy.TagsMap, event string) {
+}
+
+//Server exported
+type Server struct {
+	MutualAuth  bool
+	Validity    time.Duration
+	SecretType  tokens.SecretsType
+	ContextID   string
+	CAPEM       []byte
+	PublicPEM   []byte
+	PrivatePEM  []byte
+	rpcchannel  string
+	rpchdl      *rpcwrapper.RPCWrapper
+	StatsClient *rpcwrapper.RPCWrapper
+	Enforcer    enforcer.PolicyEnforcer
+	Collector   collector.EventCollector
+	Supervisor  supervisor.Supervisor
```

Comment:
Can you respect alphabetical order and public var first then private (same for the other struct/interface as well)